packages/headless/src/controllers/insight/facets/range-facet/numeric-facet/headless-insight-numeric-filter.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ConfigurationSection,
+  NumericFacetSection,
+  SearchSection,
+} from '../../../../../state/state-sections';
+import {loadReducerError} from '../../../../../utils/errors';
+import {
+  configuration,
+  numericFacetSet,
+  search,
+} from '../../../../../app/reducers';
+
+import {
+  NumericFilterOptions,
+  NumericFilterInitialState,
+  NumericFilterRange,
+  NumericFilterProps,
+  NumericFilterState,
+  NumericFilter,
+  buildCoreNumericFilter,
+} from '../../../../core/facets/range-facet/numeric-facet/headless-core-numeric-filter';
+import {InsightEngine} from '../../../../../app/insight-engine/insight-engine';
+import {
+  logFacetClearAll,
+  logFacetSelect,
+} from '../../../../../features/facets/facet-set/facet-set-insight-analytics-actions';
+import {executeSearch} from '../../../../../features/insight-search/insight-search-actions';
+
+export type {
+  NumericFilterOptions,
+  NumericFilterInitialState,
+  NumericFilterRange,
+  NumericFilterProps,
+  NumericFilterState,
+  NumericFilter,
+};
+
+/**
+ * Creates a `NumericFilter` controller instance.
+ * @param engine - The headless engine.
+ * @param props - The configurable `NumericFilter` controller properties.
+ * @returns A `NumericFilter` controller instance.
```

Comment:
```suggestion
 * @returns An `InsightNumericFilter` controller instance.
```

Reply:
I'll keep `NumericFilter` to match the function signature.